service/mapper/fsm_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package mapper
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/optakt/flow-dps/testing/mocks"
+)
+
+func TestNewFSM(t *testing.T) {
+	st := &State{}
+
+	t.Run("nominal case without options", func(t *testing.T) {
+		t.Parallel()
+
+		f := NewFSM(st)
+
+		assert.NotNil(t, f)
+		assert.Equal(t, st, f.state)
+	})
+
+	t.Run("nominal case with transition option", func(t *testing.T) {
+		t.Parallel()
+
+		f := NewFSM(st, WithTransition(StatusEmpty, func(_ *State) error { return nil }))
+
+		assert.NotNil(t, f)
+		assert.Equal(t, st, f.state)
+		assert.NotEmpty(t, f.transitions)
+	})
+}
+
+func TestFSM_Run(t *testing.T) {
```

Comment:
Overall, it's pretty cool that you were able to test the FSM running logic too, even though I didn't consider that necessarily important.